Homework2/Task01/task01.py

```diff
@@ -0,0 +1,65 @@
+"""
+Given a file containing text. Complete using only default collections:
+    1) Find 10 longest words consisting from largest amount of unique symbols
+    2) Find rarest symbol for document
+    3) Count every punctuation char
+    4) Count every non ascii char
+    5) Find most common non ascii char for document
+"""
+import re
+import string
+from typing import List
+
+
+def open_file(file_path: str) -> str:
+    f = open(file_path)
+    return f.read()
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    text = open_file(file_path)
+    word_list = re.sub(r"[^\w\s]", "", text).split()
+    all_words = [(elem, len(elem), len(set(elem))) for elem in set(word_list)]
```

Comment:
Do you really need to precalculate lengths?

Reply:
Yes, because I sort words afterwards by these parameters. Our task says "Find 10 longest words consisting from largest amount of unique symbols", so when we choosing between 'a' and 'aa' we should take 'aa'.

Reply:
Can you calculate them during sorting?